src/MainWindow.vala

```diff
@@ -200,38 +269,140 @@ namespace PantheonCalculator {
             basic_grid.attach (button_ans, 2, 5, 1, 1);
             basic_grid.attach (button_calc, 3, 5, 1, 1);
 
-            var button_ms = new Button ("MS", _("Set memory value"));
-            button_mr = new Button ("MR", _("Recall value from memory"));
-            button_mr.set_sensitive (false);
-            var button_m_add = new Button ("M+", _("Add to stored value"));
-            var button_m_sub = new Button ("M−", _("Subtract from stored value"));
-            button_mc = new Button ("MC", _("Clear memory"));
-            button_mc.set_sensitive (false);
-            button_gt = new Button ("GT", _("Grand Total"));
-            button_gt.set_sensitive (false);
-            var button_par_left = new Button ("(", _("Start Group"));
-            var button_par_right = new Button (")", _("End Group"));
-            var button_pow = new Button ("x<sup>y</sup>", _("Exponent"));
-            button_pow.function = "^";
-            var button_sr = new Button ("√", _("Root"));
-            var button_sin = new Button ("sin", _("Sine"));
-            var button_sinh = new Button ("sinh", _("Hyperbolic Sine"));
-            var button_cos = new Button ("cos", _("Cosine"));
-            var button_cosh = new Button ("cosh", _("Hyperbolic Cosine"));
-            var button_tan = new Button ("tan", _("Tangent"));
-            var button_tanh = new Button ("tanh", _("Hyperbolic Tangent"));
-            var button_pi = new Button ("π", _("Pi"));
-            var button_e = new Button ("e", _("Euler's Number"));
-            var button_log = new Button ("log<sub>10</sub>", _("Logarithm Base 10"));
-            button_log.function = "log";
-            var button_ln = new Button ("ln", _("Natural Logarithm"));
-            var button_asin = new Button ("sin<sup>-1</sup>", _("Inverse Sine"));
-            button_asin.function = "asin";
-            var button_acos = new Button ("cos<sup>-1</sup>", _("Inverse Cosine"));
-            button_acos.function = "acos";
-            var button_atan = new Button ("tan<sup>-1</sup>", _("Inverse Tangent"));
-            button_atan.function = "atan";
-            var button_reciprocal = new Button ("x<sup>-1</sup>", _("Reciprocal"));
+            var button_ms = new Button ("MS") {
+                tooltip_text = _("Set memory value")
+            };
+
+            button_mr = new Button ("MR") {
+                action_name = ACTION_PREFIX + ACTION_BUTTON,
+                action_target = new Variant ("s", number_to_string (memory_value)),
```

Comment:
This does not work since the value of `action_target` is captured when the button is created and will not change when the button is clicked. So it always recalls `0`.

Might as well treat this button like all the other memory buttons?